[14.0][MIG][IMP] sale_order_type

BT-cserra opened the pull request:
This PR builds on https://github.com/OCA/sale-workflow/pull/1425

This PR attends the comments made on https://github.com/OCA/sale-workflow/pull/1425, including adding the commits of the two following PRs: https://github.com/OCA/sale-workflow/pull/1531 and https://github.com/OCA/sale-workflow/pull/1553. It also deletes the README so that it is re-generated.

pedrobaeza committed:
[REN] sale_order_type: Make module name singular

Commit:
[FIX] Carga de datos demo corregida.

oihane committed:
[MOD] Fixed installation error, closes #692

pedrobaeza committed:
[IMP] Expand AUTHORS

oihane committed:
[IMP] <sale_order_type> Added readme and default journal made False

pedrobaeza committed:
[FIX] sale_order_type: Some typos

sysadminmatmoz committed:
Slovene translations added

Commit:
[FIX] sale_order_type: Fixed bug when picking is created from a purchase order

yvaucher committed:
Bump version of sale_order_type

oca-transbot committed:
OCA Transbot updated translations from Transifex

oihane committed:
[FIX] <sale_order_type> Invoice state from type not applied in pickings

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] Make modules uninstallable

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

nicomacr committed:
sale_order_type

FIX case of tests

DEL refund journal field on sale order type, now same invoice journal should be used for refunds

FIX case of test

FIX sale order test

eLBati committed:
FIX sale_order_type: propagate sale type to invoice

Conflicts:
	sale_order_type/models/sale_order.py
	sale_order_type/models/stock_picking.py
	sale_order_type/tests/test_sale_order_type.py

ADD Payment Term, Pricelist and Incoterm to sale order type

Conflicts:
	sale_order_type/models/sale_order.py
	sale_order_type/tests/test_sale_order_type.py
	sale_order_type/views/sale_order_type_view.xml

ADD sale type for invoices

Conflicts:
	sale_order_type/README.rst
	sale_order_type/models/account_invoice.py
	sale_order_type/tests/test_sale_order_type.py

IMP description

nicomacr committed:
FIX test case

ADD multicompany rule for sale.order.type and company in sale.order.type (#292)

[IMP] sale_order_type: multicompany handle
Conflicts:
	sale_order_type/models/sale_order_type.py

pedrobaeza committed:
[MIG] Rename manifest files

nicomacr committed:
FIX onchange_type_id

carlosdauden committed:
sale_order_type: Manager security overwrited with same Id in Salesman security

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
sale_order_type: Remove required fields to make it more flexible

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[MIG] sale_order_type: Migrated to 10.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

chienandalu committed:
[MIG] sale_order_type: Migration to 11.0

cubells committed:
[FIX] sale_order_type: avoid error on invoices if user is only acountant

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update sale_order_type.pot

[UPD] Update sale_order_type.pot

pedrobaeza committed:
[FIX] sale_order_type: Correct parent menu

The other one is a fake one that is dangling

oca-travis committed:
[UPD] Update sale_order_type.pot

pedrobaeza committed:
[IMP] sale_order_type: Use commercial partner sale type as fallback

If a contact doesn't have sale type, we use the commercial partner one.

pedrobaeza committed:
[IMP] sale_order_type: New README style

chienandalu committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (31 of 31 strings)

Translation: sale-workflow-11.0/sale-workflow-11.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-11-0/sale-workflow-11-0-sale_order_type/es/

ivantodorovich committed:
[IMP] sale_order_type: Use 'Sales Order' config submenu (#720)

angelmoya committed:
MIG sale_order_type

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (31 of 31 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_type/zh_CN/

GSLabIt committed:
Translated using Weblate (Italian)

Currently translated at 83.9% (26 of 31 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_type/it/

nicomacr committed:
[12.0] [FIX] sale_order_type: Missing warnings on inv onchange method

In the original onchange method by odoo, this return an dict with values(warnings and domians), but when inherit this method in this module they not return this dict. Because of this the alert setting in the partner was no working.

kongrattapong committed:
[IMP] sale_order_type: black, isort

kongrattapong committed:
[MIG] sale_order_type: Migration to 13.0

pedrobaeza committed:
[FIX+IMP] sale_order_type: New computed writable + migration fixes

* Switch onchange on the main field to computed writable fields
* Record rule is not up to v13
* Added migration script for record rule
* Proper payment term field name in invoice
* Realign tests

oca-travis committed:
[UPD] Update sale_order_type.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_order_type/

ivantodorovich committed:
[IMP] Added route field in sale type

ivantodorovich committed:
[IMP] Move sale_order_type to Sales group

ivantodorovich committed:
[IMP] sale_order_type: Add missing fields in view and use groups to hide unnecessary features

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update sale_order_type.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_order_type/

ivantodorovich committed:
[FIX] Properly display sale_type_id field in account.move form view.

ivantodorovich committed:
[IMP] Do not allow to edit sale_order_type on confirmed records

OCA-git-bot committed:
sale_order_type 13.0.1.2.0

sergio-teruel committed:
[FIX+IMP] sale_order_type: Proper computed writable definition + field position + improve test

- A computed writable field is not that for the ORM if no readonly=False, so we have
  to inverse current conditions.
- Field is not properly viewed in the move form in previous position.
- Tests are more bullet-proof now creating real records with Form and with a little
  code reorganization.

danimv5 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (32 of 32 strings)

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_order_type/es/

grindtildeath committed:
sale_order_type: add default value to fix breaking tests

As the field is set as required on the Form view, any other module
that does not depend on sale_order_type and that uses test class
Form on sale.order would fail with "type_id is a required field"
if no default value is defined.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_order_type/

nicomacr committed:
[13.0] [FIX] sale_order_type: Only show the type field in sales

Actualy the field type is showed in all of the invoice types (customer and vendors), but the for the vendors invoices, it's not necesary to show it . Also the same apply for the compute method for that field.

OCA-git-bot committed:
sale_order_type 13.0.1.3.0

sergio-teruel committed:
[FIX] sale_order_type: Use record instead of self

brendapaniagua committed:
Translated using Weblate (Catalan)

Currently translated at 93.8% (30 of 32 strings)

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_order_type
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_order_type/ca/

pedrobaeza committed:
[FIX] sale_order_type: Proper no partner field

It's type_id, not sale_type_id.

Fixes #1315

nicomacr committed:
[13.0] [FIX] sale_order_type: Not allowed to delete a sale order type if them is used in an sale order.

So keep in mind that if you accidentally delete a sales type, it is protected until you remove it from the sales order.
In case you need to perform an operation such as printing the report and the sales order is already confirmed. Because the field is requiered only per view.

OCA-git-bot committed:
sale_order_type 13.0.1.3.2

pedrobaeza committed:
[FIX] sale_order_type: Copy type in sales orders

If not, the default is applied, and it doesn't make sense in the duplication context, as
the logical one is to preserve the type of the duplicated order.

OCA-git-bot committed:
sale_order_type 13.0.1.3.4

pedrobaeza committed:
[FIX] sale_order_type: Preserve type on move copy

Same as 536301a2ac5c8c537a3cfe3a748436a60bbca5f3, but for moves.

pedrobaeza committed:
[FIX] sale_order_type: Restrict order type search to current company

A user can be active in multiple companies, when searching for a sale order type when
none is set, this should be taken into account.

Closes #1355

newtratip committed:
[IMP] sale_order_type: black, isort, prettier

newtratip committed:
[MIG] sale_order_type: Migration to 14.0

chandni300 committed:
[12.0][IMP]sale_order_type: update quotation name when user change order type

GSLabIt committed:
[12.0][FIX] sale_order_type, singleton error on write

nicomacr committed:
[FIX] sale_order_type: Non reset the value when it's sitting by the user in the view.

In this way we reset so that if I choose a value in the view this value is not stepped on by the compute method, before the change it always stepped on regardless that there are any value to replace about on the outcome of the logic.

chienandalu committed:
[FIX] sale_order_type: changes ignored

f4e61607fae6f735452b735a4cd7f45c5914f71e added some changes that made
imposible to reach a proper write if the sale type didn't have the
proper conditions.

BT-cserra committed:
[IMP] Attend code review comments

- Use of self.env.context instead of the legacy self._context
- Name the folder containing the wizards `wizards`

BT-cserra committed:
Delete the README so that it is regenerated

After adding the following hotfix PRs, the readme was
outdated. I remove it so that OCA regenerates it:
- https://github.com/OCA/sale-workflow/pull/1531
- https://github.com/OCA/sale-workflow/pull/1553

oihane reviewed (approved):
LGTM

ioans73 commented on sale_order_type/models/sale.py:
```diff
@@ -0,0 +1,115 @@
+        if vals.get("name", "/") == "/" and vals.get("type_id"):
+            sale_type = self.env["sale.order.type"].browse(vals["type_id"])
+            if sale_type.sequence_id:
+                vals["name"] = sale_type.sequence_id.next_by_id()
```
I think this module is not working properly with the subsequences by date range.
I configure the subsequences for sale orders and, when I create once for the next year, it assings the 2021 sequence.

![sale_order_type_range_sequence](https://user-images.githubusercontent.com/29223264/124739060-ceb3d300-df19-11eb-8971-c13a8866b181.gif)

ioans73 replied:
ping @BT-cserra

BT-cserra replied:
@ioans73 did you test with V13 or with this PR?

ioans73 replied:
@BT-cserra with this PR

BT-cserra replied:
Many thanks @ioans73 you were 100% right! There was a missing parameter. [This commit](https://github.com/OCA/sale-workflow/pull/1646/commits/d53ccc6af7a5d70910c01c28bd49d843dc58d523) fixes it. I've requested a re-review.

BT-cserra replied:
Wait @ioans73 I forgot to pass the tests. Ouch.

BT-cserra replied:
I force pushed with https://github.com/OCA/sale-workflow/pull/1646/commits/d53ccc6af7a5d70910c01c28bd49d843dc58d523. Now the checks are in green.

ioans73 replied:
Perfect, thank you very much @BT-cserra

BT-cserra committed:
[14.0][FIX] sale_order_type: sequence depending on date ranges

The call to obtain the next sequence number for a sales order
didn't take into account its date, so as a result the option
to use a different sequence depending on the date was ignored.

newtratip commented:
@BT-cserra, ready to merge ?

BT-cserra commented:
> @BT-cserra, ready to merge ?

@newtratip yes.

newtratip commented:
/ocabot merge nobump
